### ISSUE TITLE
Replacing CLIP and InternVideo2 with DeepSeek's Janus models in an image-based version of GenRL

### ISSUE THREAD

**Issue by ajaytalati** [mazpie/genrl#13]
Hi Pietro,

I really like the GenRL paper! 

I was wondering if it would be possible to create a simpler version of GenRL (GenRL-lite), which would be more suitable for low resource training from scratch? In the paper it says GenRL took 5 days to train on a cluster of V100! To understand GenRL better, would it be possible to replace CLIP and InternVideo2 with DeepSeek's Janus models in an image-based version of GenRL? Once trained, could GenRL-lite be used to try to reproduce the Robo-Yoga benchmark?

Best regards,

Ajay

PS - I think DeepSeek's Janus is promising as if I've understood correctly it uses vector-quantization. Also, it seems to have an understanding of how an agent like "stickman" should act on their joint angles, to move from an "initial state" to a "goal-state"? 

I used the HuggingFace link [https://huggingface.co/spaces/deepseek-ai/Janus-Pro-7B](https://huggingface.co/spaces/deepseek-ai/Janus-Pro-7B), and uploaded this image,

![Image](https://github.com/user-attachments/assets/31732575-50c5-4025-800f-8e7423ea146b)

I asked the question/prompt, 

"Which yoga pose is this? Lets call the final position the "goal state". Give instructions how a human should move their joints into this goal state, from the start here "initial state"

It replied,

"This yoga pose is called "Headstand" or "Upward-Facing Dog" in Sanskrit. Here are the instructions to move into the goal state:

1. **Start Here:**
   - Begin on your hands, with your hands shoulder-width apart on the mat.
   - Engage your core and keep your spine straight.
   - Place your feet flat on the mat, with your knees slightly bent if necessary.

2. **Move to Get Here:**
   - From the start position, inhale and press your hands into the mat.
   - Exhale and lift your hips up and back, creating a straight line from your head to your heels.
   - Keep your arms straight and your shoulders down.
   - Ensure your core muscles are engaged to maintain stability.

3. **Final Goal State:**
   - In the goal state, your body should form a straight line from head to heels.
   - Your arms should be straight, and your shoulders should be relaxed.
   - Your head should be directly above your hands, and your gaze should be forward.
   - Keep your core muscles engaged to maintain balance."

It seems like it understands sequences of images, and how they should encoded/mapped to actions?

**Comment by mazpie** [comment]
Hi @ajaytalati ,

I am not fully familiar with the Janus model yet. Overall, I see that the model it's not trained with contrastive pre-training (like CLIP or InternVideo). However, the semantic decoder is trained to reconstruct SigLIP features. 

Perhaps, the output of the semantic decoder could be used for your purpose.

P.S. I think Janus hallucinated a bit, as that yoga pose shouldn't be called the "Upward-Facing Dog". Headstand should be the correct label (the model mentioned both). This could be confusing for an action model.

**Comment by ajaytalati** [comment]
Hi Pietro,

I've spent quite a long time trying to see if Janus, or any other model with language capabilities understands the sequence of  muscle activation's and limb movements for yoga asanas and even some basic physiotherapy exercises. Seems like they just regurgitate patterns in the text they have memorized without any bio-mechanical understanding. Many of the instructions they output are nonsense? 

I don't think Janus would be any better than CLIP + InternVideo2.

**Comment by mazpie** [comment]
Thank you for the valuable insights!

I look forward to seeing models with better capabilities to integrate with GenRL's recipe in the future!
